### PR TITLE
Leaderboard: API implementation

### PR DESCRIPTION
**Pull request by d4nQw3rty**
This branch will implement the API for the leaderboard app:
### Requirements:
- Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- Make sure that you used the correct flow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
- Follow Microverse's list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).
- Create a new game with the name of your choice by using the API.
- Implement the "Refresh" button that will get all scores for a game created by you from the API (receiving data from the API and parsing the JSON).
- Implement the form "Submit" button that will save a score for a game created by you (sending data to the API).
- Use arrow functions instead of the function keyword.
- Use async and await JavaScript features to consume the API.
- No styling is required.

### CODE REVIEW

File: src/modules/scoresMethods.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    return response.json();
+  },
+  async getScores() {
+    const idUrl = 'https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/x7gdzdyghc93DAWobtyz/scores/';
+    const response = await (await fetch(idUrl)).json();
+    return response.result;
+  },
+};
+
+export default easterEggQuest;
```
Comment (HAMMAS-SALEEM) [inline src/modules/scoresMethods.js:23]: - [x] Its great that you have implemented the functionality to add data to the API :blush: but when i click on the refresh button then it doesn't give me the new data which i have added kindly fix it :+1:
- [x] Your page is not supposed to reload by clicking on the Refresh button of data. Kindly get the data without refreshing the page while clicking on the Refresh button :+1:

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @d4nQw3rty :wave:

## Highlights :1st_place_medal:

:heavy_check_mark: Descriptive PR Title
:heavy_check_mark: Descriptive PR Summary
:heavy_check_mark: Following Linters
:heavy_check_mark: Following some requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/scoresMethods.js (hunk 1 above)
Comment (d4nQw3rty) [reply src/modules/scoresMethods.js:23]: Hello, @HAMMAS-SALEEM  Thanks for the review I can now improve my code. :+1: 
It was my fault for not telling you that it was a top 10, and the Maximus score was 100, you submitted 101 twice; now I know. :smile:
But I fixed it now the leaderboard accepts 1000 participants and the Maximus score is 1000. :100: 
Now the page displays the leaderboard and doesn't reload the URL.

**Review by omarsalem7 — APPROVED** [review]
Hi @d4nQw3rty 😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Descriptive pull request 
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**